types/extract-files/index.d.ts

```diff
@@ -1,20 +1,32 @@
-// Type definitions for extract-files 3.1
+// Type definitions for extract-files 8.1
 // Project: https://github.com/jaydenseric/extract-files#readme
 // Definitions by: Edward Sammut Alessi <https://github.com/Slessi>
+//                 Alex K <https://github.com/lynxtaa>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+export type ExtractableFileMatcher = (value: any) => boolean;
+
+export const isExtractableFile: ExtractableFileMatcher;
+
+export function extractFiles<T>(
+    value: T,
+    path?: string,
+    isExtractableFile?: ExtractableFileMatcher,
+): {
+    clone: T;
```

Comment:
@lynxtaa https://github.com/jaydenseric/extract-files#examples-1

What I understood from README is that `value` and `clone` are _not_ the same, but you have them both as `T` - can you explain?

> ```ts
> const value = {
>   a: file1,
>   b: [file1, file2],
> };
> 
> const { clone, files } = extractFiles(value, 'prefix');
> ```
> `value` remains the same.
> 
> `clone` is:
> 
> ```ts
> {
>   "a": null,
>   "b": [null, null]
> }
> ```